### PR TITLE
Adding external Force Density to grains

### LINKED ISSUE #5521: Adding External Force Density to grains
Adds external force density to grains in addition to reaction forces due to interaction between grains. This force will be used in grain motion kernels as well as by mechanics modules for stress calculation. This should be equivalent to applying pressure boundary conditions for cases combining PhaseField and TensorMechanics. Helpful for grains in porous media.

### PR DESCRIPTION
**Pull request by SudiptaBiswas**
Adds ExternalForceDensity material, modifies grain advection materials and kernels.

Closes #5521

### CODE REVIEW

File: modules/phase_field/src/materials/GrainAdvectionVelocity.C [hunk 1]
```diff
@@ -14,6 +14,7 @@ InputParameters validParams<GrainAdvectionVelocity>()
   params.addCoupledVar("etas", "Array of other coupled order parameters");
   params.addParam<Real>("translation_constant", 500, "constant value characterizing grain translation");
   params.addParam<Real>("rotation_constant", 1.0, "constant value characterizing grain rotation");
+  params.addParam<std::string>("force_type", "Optional parameter that allows the user to define type of force density under consideration");
```
Comment (dschwen) [inline modules/phase_field/src/materials/GrainAdvectionVelocity.C]: I'd prefer we stick to ```base_name``` here for consistency.

File: modules/phase_field/src/userobjects/ComputeGrainForceAndTorque.C [hunk 2]
```diff
@@ -13,6 +13,7 @@ InputParameters validParams<ComputeGrainForceAndTorque>()
   InputParameters params = validParams<ElementUserObject>();
   params.addClassDescription("Userobject for calculating force and torque acting on a grain");
   params.addParam<MaterialPropertyName>("force_density", "force_density", "Force density material");
+  params.addParam<MaterialPropertyName>("force_density_derivative", "dFdc", "Derivative of force density material");
```
Comment (dschwen) [inline modules/phase_field/src/userobjects/ComputeGrainForceAndTorque.C]: We should convert this to the material property derivative system (look at ```DerivativeMaterialPropertyNameInterface```)

**Comment by jwpeterson** [comment]
These 5 commits should be squashed into 2 before merge (one for new code, one for new tests).  @dschwen can assist with that if needed.

**Comment by SudiptaBiswas** [comment]
I will take care of the commits.

**Pull request merged**